thingsboard_gateway/connectors/modbus_async/slave.py

```diff
@@ -0,0 +1,233 @@
+import asyncio
+from threading import Thread
+from time import sleep, monotonic
+
+from pymodbus.constants import Defaults
+
+from thingsboard_gateway.connectors.modbus_async.bytes_modbus_downlink_converter import BytesModbusDownlinkConverter
+from thingsboard_gateway.connectors.modbus_async.bytes_modbus_uplink_converter import BytesModbusUplinkConverter
+from thingsboard_gateway.connectors.modbus_async.configs.bytes_downlink_converter_config import \
+    BytesDownlinkConverterConfig
+from thingsboard_gateway.connectors.modbus_async.configs.bytes_uplink_converter_config import BytesUplinkConverterConfig
+from thingsboard_gateway.gateway.constants import UPLINK_PREFIX, CONVERTER_PARAMETER, DOWNLINK_PREFIX
+from thingsboard_gateway.gateway.statistics.statistics_service import StatisticsService
+from thingsboard_gateway.tb_utility.tb_loader import TBModuleLoader
+
+
+class Slave(Thread):
+    def __init__(self, **config):
+        super().__init__()
+        self.daemon = True
+        self.stopped = False
+        self._log = config['logger']
+        self.connector = config['connector']
+        self.name = "Modbus slave processor for unit " + str(config['unitId']) + " on host " + str(
+            config['host']) + ":" + str(config['port'])
+
+        self.callback = config['callback']
+
+        self.unit_id = config['unitId']
```

Comment:
Such as you have added “configs”, why you cannot add one more abstract class config, and two child classed - SerialSlaveConfig and TcpSlaveConfig and move related properties, callback, etc there?

Reply:
Also, you can add method validate to configuration entity, validate configuration after creation and avoid creating of the slave if the configuration is bad.